KIKIMR-26698 topics doc fixes (#21099)
ydb-platform/ydb: ydb/docs/en/core/concepts/topic.md

When several application instances read messages from a stream, a message about account top-ups can be received by one instance and a message about debiting by another. In this case, there's no guaranteed instance with accurate balance information. To avoid this issue, you can, for example, save data in the DBMS, share information between application instances, and implement a distributed cache.

> {{ ydb-short-name }} can write data so that messages from one source (for example, about transactions from one account) arrive at the same application instance. Each source writes messages with its own unique producer_id, and a sequence number (seqno) is used to prevent duplicate processing. {{ ydb-short-name }} routes all messages with the same producer_id to the same partition. As a result, transaction messages for a given account always arrive at the same partition and are processed by the application instance associated with it. Each instance handles its own subset of partitions, so there is no need for synchronization between instances.

Reviewer: ```suggestion
{{ ydb-short-name }} can write data so that messages from a single source are delivered to the same application instance. Each source writes messages with its own unique `producer_id`, and a sequence number (`seqno`) is used to prevent duplicate processing. {{ ydb-short-name }} routes all messages with the same `producer_id` to the same partition. Each instance handles its own subset of partitions, eliminating the need for synchronization between instances. For example, this approach could allow all transactions from a given account to be processed by the application instance associated with it. 
```

For some tasks, the message processing order is not critical. For example, it's sometimes important to simply deliver data that will then be ordered by the storage system.

> For such tasks, the 'no-deduplication' mode can be used. In this scenario [`producer_id`](#producer-id) doen't specified in write session setup and [`sequence numbers`](#seqno) are also not used for messages. The no-deduplication mode offers better performance and requires fewer server resources, but there is no message ordering or deduplication on the server side, which means that a message sent to the server multiple times (for example, due to network instability or writer process crash) may also be written to the topic multiple times.

Reviewer: ```suggestion
For such tasks, the "no-deduplication" mode can be used. In this scenario, [`producer_id`](#producer-id) isn't specified in the write session setup, and [`sequence numbers`](#seqno) aren't used for messages. The no-deduplication mode offers better performance and requires fewer server resources; however, there is no message ordering or deduplication on the server side. This means that a message sent to the server multiple times (for example, due to network instability or a writer process crash) may be written to the topic multiple times.
```
